### ISSUE TITLE
Tutorial should discuss lexers

### ISSUE THREAD

**Issue by Timmmm** [zesterer/chumsky#227]
Hi, I've been meaning to try Chumsky for ages and finally got round to reading the tutorial. Mostly makes sense. Then I look at the examples, e.g. `nano_rust.rs` and it's quite different to the code in the tutorial - it has a separate lexer and parser whereas the tutorial lexes as it parses.

I wonder if it is worth mentioning in the tutorial why this is done? Presumably there are some advantages to it but it isn't obvious what they are just from reading the code (unless it's just easier to organise that way).

**Comment by zesterer** [comment]
Yes, I agree. I've been meaning to go into a little more detail about lexing and perhaps separate out the tutorial into a separate lexing stage. I've not yet had time to commit to this though.

**Comment by Zij-IT** [comment]
This could be a good candidate for a chapter in the book that I am trying to introduce with #442.
